src/ping.rs

```diff
@@ -121,17 +121,18 @@ pub fn send_pings(
             }
             ping.seen = false;
         }
+        let start_time = Instant::now();
         {
             // start the timer
             let mut timer = timer.write().unwrap();
-            *timer = Instant::now();
+            *timer = start_time;
         }
         loop {
             // use recv_timeout so we don't cause a CPU to needlessly spin
             match thread_rx
                 .lock()
                 .unwrap()
-                .recv_timeout(Duration::from_millis(100))
+                .recv_timeout(max_rtt.saturating_sub(start_time.elapsed()))
```

Comment:
Am I understanding this correctly that `saturating_sub` essentially clamps the value?

Reply:
Yes, that is correct:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=346d79c451e079ae1da20226f59b6308